executor/index_lookup_join.go

```diff
@@ -392,9 +392,6 @@ func (iw *innerWorker) run(ctx context.Context, wg *sync.WaitGroup) {
 
 		err := iw.handleTask(ctx, task)
 		task.doneCh <- errors.Trace(err)
-		if err != nil {
```

Comment:
we should close `iw.taskCh` to let outer workers stop running as soon as possible.

Reply:
When the caller received the error, it will close the executor, then the context will be canceled, all workers will quit.